app/views/admin/menu/_menu.html.erb

```diff
@@ -26,7 +26,9 @@
   <%= render partial: 'admin/menu/menu_other' %>
   <% end %>
 
-  <% if current_user.can? :list, :web_stats %>
+  <% if current_user.can?( :view_web,   :stats ) ||
```

Comment:
To consolidate these `current_user.can?` checks into as few templates as possible, and so this responsibility isn't spread so much across templates, consider moving the checks from `_menu` to the top of the `_menu_stats` partial and doing an early `return` if they're unauthorised:

```erb
<% return unless current_user.can?(:view_web, :stats) || ... || ... %>
```

Reply:
Ooh, I didn't realise you could return early from templates. That's handy.